Kernel/Prekernel/Arch/x86/boot.S

```diff
@@ -104,108 +104,28 @@ start:
 */
 print_and_halt:
 
-/*  from now on, we don't really care about booting because we are missing required CPU features such as PAE or long mode.
-    the flow from now is like so:
-    1. Copy all necessary parts to low memory section in RAM
-    2. Jump to that section
-    3. In that section we do:
-        a. exit protected mode to pure 16 bit real mode
-        b. load the "<missing feature> is not supported" String, call the BIOS print to screen service
-        c. halt
-*/
-
-.equ COPIED_STRING_LOCATION, 0x400
-.equ GDT_REAL_MODE_LOCATION, 0x45000
-.equ EXITING_PROTECTED_MODE_CODE_LOCATION, 0x10000
-.equ REAL_MODE_CODE, 0x500
-.equ PROTECTED_MODE_16_BIT_CODE, 0x600
     movl %esp, %ebp
-    movl 4(%ebp), %edi
+    movl 4(%ebp), %esi
 
-    /* Copy string to low memory section  */
-    movl %edi, %esi
-    xor %ecx, %ecx
+    mov $0xb8000, %ecx    /* VRAM address. */
```

Comment:
Although technically correct as an implementation, how do we know that the bootloader left us with text mode to work with?

Reply:
I guess we can't really check it, unless we drop into 16 bit code to ask the BIOS.

We can only assume that the bootloader dropped us into text mode, which most should be doing. And even if we are not currently in text mode, writing to that address shouldn't break too much and the now restored BIOS interrupt should still have a chance to write his message.